packages/app/src/cli/services/function/replay.test.ts

```diff
@@ -87,54 +107,64 @@ const UNRELATED_LOG: FunctionRunData = {
   log_timestamp: '2024-05-31T15:29:50.741270Z',
   identifier: '123456',
 }
-const UNRELATED_LOG_FILENAME = UNRELATED_LOG.log_timestamp.replace(/:/g, '_')
+
+const API_KEY = 'apiKey'
+
+async function writeLogFile(options) {
+  const index = options.index || 0
+  const handle = options.handle || HANDLE
+  const filename = options.run.log_timestamp.replace(/:/g, '_')
+  return writeFile(
+    joinPath(options.runDir, `${filename}_extensions_${handle}_${index}_${options.run.identifier}.json`),
+    JSON.stringify(options.run),
+  )
+}
+
+async function replayWithMocks(tmpDir, functionRunsDir) {
+  const app = testApp()
+  const developerPlatformClient = testDeveloperPlatformClient()
+  //
+
+  vi.mocked(ensureConnectedAppFunctionContext).mockResolvedValueOnce({apiKey: API_KEY, developerPlatformClient})
+  vi.mocked(getLogsDir).mockReturnValue(tmpDir)
+  vi.mocked(selectFunctionRunPrompt).mockResolvedValue(RUN1)
+  vi.mocked(runFunctionRunner)
+
+  const options = {
+    app,
+    extension: EXTENSION,
+    apiKey: undefined,
+    stdout: false,
+    path: functionRunsDir,
+    json: true,
+  }
+
+  return replay(options)
+}
```

Comment:
I extracted this code from the tests since I was finding it difficult to separate out the implementation of the test from what the test was actually testing- I think the resulting tests are easier to parse, but let me know what you think